Modules/cjkcodecs/multibytecodec.c

```diff
@@ -1119,6 +1197,82 @@ _multibytecodec_MultibyteIncrementalDecoder_decode_impl(MultibyteIncrementalDeco
     return NULL;
 }
 
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalDecoder.getstate
+[clinic start generated code]*/
+
+static PyObject *
+_multibytecodec_MultibyteIncrementalDecoder_getstate_impl(MultibyteIncrementalDecoderObject *self)
+/*[clinic end generated code: output=255009c4713b7f82 input=4006aa49bddbaa75]*/
+{
+    PyObject *buffer;
+    unsigned long long flag;
+
+    if (self->pendingsize > 0) {
+        buffer = PyBytes_FromString((const char *)self->pending);
+    } else {
+        buffer = PyBytes_FromString("");
+    }
+    if (buffer == NULL) {
+        return NULL;
+    }
+
+    flag = self->state.ull;
+
+    return Py_BuildValue("NK", buffer, flag);
+}
+
+/*[clinic input]
+_multibytecodec.MultibyteIncrementalDecoder.setstate
+    state: object
+    /
+[clinic start generated code]*/
+
+static PyObject *
+_multibytecodec_MultibyteIncrementalDecoder_setstate(MultibyteIncrementalDecoderObject *self,
+                                                     PyObject *state)
+/*[clinic end generated code: output=26556d397fc504d8 input=1d31c7a7b0b04371]*/
+{
+    PyObject *buffer;
+    PyObject *ret;
+    Py_ssize_t buffersize;
+    char *bufferstr;
+    unsigned long long flag;
+
+    if (!PyTuple_Check(state)) {
+        PyErr_SetString(PyExc_TypeError, "state argument must be a tuple");
```

Comment:
PyArg_ParseTuple will raise TypeError.  So this check may be redundant.

Reply:
PyArg_ParseTuple is enough but the error message is not very user friendly:

```
>>> import codecs
>>> decoder = codecs.getincrementaldecoder('euc_jp')()
>>> decoder.setstate(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
SystemError: new style getargs format but argument is not a tuple
```

Reply:
Then, you can use ArgumentClinic again.  `object(subclass_of='&PyTuple_Type')`.

Reply:
Thanks, that works nicely :)

```
>>> decoder.setstate(0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: setstate() argument must be tuple, not int
```